os/fs/driver/mtd/mtd_partition.c

```diff
@@ -427,10 +428,12 @@ static int part_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg)
 
 	case MTDIOC_BULKERASE: {
 		/* Erase the entire partition */
-
-		ret = priv->parent->erase(priv->parent, priv->firstblock / priv->blkpererase, priv->neraseblocks);
-		if (ret == priv->neraseblocks) {
-			ret = OK;
+		FAR struct mtd_geometry_s *geo = (FAR struct mtd_geometry_s *)arg;
+		if (geo) {
+			ret = priv->parent->ioctl(priv->parent, MTDIOC_BULKERASE, (unsigned long)((uintptr_t)geo));
```

Comment:
Here, the return value corresponds directly to the return value of imxrt function. It should be differentiated.

Reply:
You're right, thx.
Modified.